doc/melange.rst

```diff
@@ -0,0 +1,296 @@
+.. _melange_main:
+
+***********************************
+JavaScript Compilation With Melange
+***********************************
+
+Introduction
+------------
+
+`Melange <https://github.com/melange-re/melange>`_ is a compiler from OCaml to
+JavaScript. Unlike Js_of_ocaml, Melange works by translating OCaml compiler
+internal Lambda representation to JS files. This allows to produce a single
+JavaScript file from each OCaml module. Melange can be installed with
+`opam, Esy, or Nix <https://github.com/melange-re/melange#installation>`_
+package managers.
+
+Dune can build projects using Melange, and it allows the user to generate
+JavaScript files by defining a :ref:`melange-emit` stanza. Dune libraries can be
+used with Melange by leveraging the ``(modes melange)`` field in the
+:ref:`library` stanza.
+
+To enable Melange support in a Dune project, specify the Melange language
+version in the :ref:`dune-project` file. Adding
+
+.. code:: scheme
+
+    (using melange 0.1)
+
+to a :ref:`dune-project` file enables using the ``melange.emit`` stanza and the
+``(modes melange)`` field for the :ref:`library` stanza.
+
+.. _melange-emit:
+
+melange.emit
+------------
+
+The ``melange.emit`` stanza allows the user to generate JavaScript from Melange
+libraries and modules. It's similar to the OCaml :ref:`executable` stanza, but
+for Melange projects:
+
+.. code:: scheme
+
+    (melange.emit
+     (target <target>)
+     <optional-fields>)
+
+``<target>`` is the name of the folder where resulting JavaScript artifacts will
+be placed. In particular, the folder will be placed under
+``_build/default/$path-to-directory-of-melange-emit-stanza``.
+
+The result of building a  ``melange.emit`` stanza will match the file structure
+of the source tree. For example, given the following source tree:
+
+.. code:: bash
+
+    ├── dune # (melange.emit (target output) (libraries lib))
+    ├── app.ml
+    └── lib
+        ├── dune # (library (name lib) (modes melange))
+        └── helper.ml
+
+The resulting layout in ``_build/default/output`` will be as follows:
+
+.. code:: bash
```

Comment:
```suggestion
.. code::
```

Reply:
@emillon Using `bash` allows to show the comments in gray:

<img width="721" alt="imagen" src="https://user-images.githubusercontent.com/220424/229815193-3cb4c81a-f93f-4971-b833-3207b42b7a7f.png">


vs without bash

<img width="728" alt="imagen" src="https://user-images.githubusercontent.com/220424/229815289-555bf4bb-ba41-473c-ae07-145900a81d1b.png">

Isn't the former better?

Reply:
Oh, I see now the suggestion was for a block without comments 😅 Will update.